### ISSUE TITLE
set :filter, :roles => %w{web} not working

### ISSUE THREAD

**Issue by joelataylor** [capistrano/capistrano#1457]
Hi, in referencing this page: http://capistranorb.com/documentation/advanced-features/role-filtering/
I would assume putting `set :filter, :roles => %w{web}` in my deploy.rb file it would work (by only deploying to the servers listed in the web role. However, it doesn't work.

To get it to work I have to use either command line: `ROLES=web cap production deploy` or `cap --roles=web production deploy`. 

I'm using 3.4.0

**Comment by joelataylor** [comment]
Anyone else having this issue?

**Comment by leehambley** [comment]
You can work around it by specifying the CLI options instead, I have yet to investigate this.

**Comment by townsen** [comment]
Looks like there is a typo in the doc - it should be 'role' singular, not 'roles':
```
set :filter, :role => %w{web}
```
Can you try this?

**Comment by townsen** [comment]
@leehambley On looking at the code there are tests in place for this and they also refer to 'role' and 'host', so it was a doc error. Although I'm tempted to change the names to 'roles' and 'hosts' to fit in with the command line and ENV names, doing so would break existing usage. How about we allow both? I'll create a PR that just makes this addition, then we can leave the documentation unchanged, and existing users won't break.

**Comment by joelataylor** [comment]
@townsen ahh yeh, that makes sense. Although I found another work-around (by setting up my servers with extra params, ie: `no_release`) I think your solution sounds perfect. Nice to see the PR so quickly! Good work. :100:

**Comment by leehambley** [comment]
@joelataylor thanks to @townsen 's work you can now use Capistrano from master, simply change:

``` diff
- gem "capistrano"
+ gem "capistrano", github: "capistrano/capistrano"
```

in your gemfile, I hope to get the release out including this fix before the end of play today.

**Comment by joelataylor** [comment]
Thanks @leehambley & @townsen - glad we could fix a bug quickly. :)

**Comment by marzapower** [comment]
The official pages are not aligned with this issue: http://capistranorb.com/documentation/advanced-features/role-filtering/

I am using
`set :filter, :roles => %w{app web}`
as stated, but it will run the deploy chain on all the other servers too.

If I use
`set :filter, :role => %w{app web}`
everything works fine.

Is this wanted?

**Comment by leehambley** [comment]
We've had some issue with `role` vs `roles` being accepted in the keys - I think we decided that the docs are out of date, since there's people with code in the wild.

Personally it's a feature I never use, so it's hard to say what it *should* be, but I think `code > docs` in terms of deciding for A or B.

**Comment by mattbrictson** [comment]
@marzapower `:roles` didn't work before, but it was fixed in 3.5.0: #1465

If it still doesn't work for you in 3.5.0, we'll reopen this issue.

**Comment by marzapower** [comment]
Sorry, it was my fault. I'm still on 3.4.x, I haven't yet updated to 3.5.0. No need to reopen the issue.
